### PR TITLE
Add stat_ecdf support

### PR DESCRIPTION
**Pull request by moutikabdessabour**
- [x] change `NEWS.md`

### CODE REVIEW

File: R/layers2traces.R [hunk 1]
```diff
@@ -623,7 +623,8 @@ to_basic.GeomQuantile <- function(data, prestats_data, layout, params, p, ...){
 }
 
 #' @export
-to_basic.default <- function(data, prestats_data, layout, params, p, ...) {
+to_basic.default<- function(data, prestats_data, layout, params, p, ...) {
+
```
Comment (cpsievert) [inline R/layers2traces.R:627]:
```suggestion
to_basic.default <- function(data, prestats_data, layout, params, p, ...) {
```

File: R/layers2traces.R [hunk 2]
```diff
@@ -252,7 +252,7 @@ to_basic.GeomLine <- function(data, prestats_data, layout, params, p, ...) {
 
 #' @export
 to_basic.GeomStep <- function(data, prestats_data, layout, params, p, ...) {
-  prefix_class(data, "GeomPath")
+  prefix_class(if((params$direction %||% "vh")  %in% c("vh", "hv", "mid")) ggplot2:::stairstep(data, direction = params$direction) else data, c("GeomPath", "GeomStep"))
```
Comment (cpsievert) [inline R/layers2traces.R:255]: Using `:::` isn't allowed by `R CMD check`. Use `ggfun()` instead

Comment (cpsievert) [reply R/layers2traces.R:255]: For some reason this breaks default `geom_step()` usage. See, for example, the change in `tests/testthat/_snaps/ggplot-step/step-gg-hv.svg`

Comment (moutikabdessabour) [reply R/layers2traces.R:255]: I think I fixed the test issues. basically just re rendering the scg solved it

Comment (cpsievert) [reply R/layers2traces.R]: No, the tests are still broken. I don't think this is the right fix.

**Comment by cpsievert** [comment]
This isn't the right approach

**Pull request closed without merging**